core/src/main/scala-3/neotypes/CypherStringInterpolator.scala

```diff
@@ -0,0 +1,129 @@
+                .asExprOf[QueryArg]
+              '{ Right($e) }
+
+            }
+            case isf: ImplicitSearchFailure =>
+              val e = Expr(isf.explanation)
+              '{ compiletime.error($e) }
+          }
+
+    def parts: List[Expr[Boolean => Res]] = {
+      val inverted = invert(argsExpr)
+      inverted
+        .map { (expr: Expr[Any]) =>
+          lazy val qarg: Expr[Res] = asRightQueryArg(expr)
+          '{ (b: Boolean) =>
+            if (b)
+              $qarg
+            else Left($expr.toString)
+          }
+        }
+    }
+
+    val partsPacked: Expr[List[Boolean => Res]] = Expr.ofList(parts)
+
+    val stringFragmentsPartitionedByInterpolationKind: Expr[List[Either[String, String]]] = '{
+      $clz
+        .parts
+        .map { part =>
+          if (part.endsWith("#"))
+            // 1. string before interpolation: "foo #${} bar" or "foo bar#${}"
+            //    In this case, `#` is a marker for interpolation that should be dropped.
+            // 2. the hashtag at the end of string: "foo bar#"
+            //    It is unusual that cypher query ends with `#`
+            //    so, optimistically use `part.init` here.
+            Left(
+              part.init
+            ) // Left means "nth part is `part.init` and the interpolation after the nth part is a plain value"
+          else
+            // Right means "nth part is`part` and the interpolation after the nth part (if any) is a query parameter"
+            Right(part)
+        }
+        .toList
+    }
+    val interleaved: Expr[List[Res]] =
+      '{
+        Interleave($partsPacked, $stringFragmentsPartitionedByInterpolationKind)
+      }
+
+    '{
+      createQuery($interleaved: _*)
+    }
+}
+
+object Interleave {
+  import CypherStringInterpolator.Res
+
+  /** @param interpolations
+    *   a list of interpolation generator functions. If a function receives `true` it should generate QueryArg
+    *   representation and otherwise it should generate String representation.
+    * @param stringFragments
+    *   a list of string splitted by interpolation and distinguished by interpolation kind. If the n-th element of
+    *   `stringFragments` is `Right`, it implies the next adjacent interpolation should be a `QueryArg` interpolation.
+    *   If the n-th element of `stringFragments` is `Left`, it implies the next adjacent interpolation should be a plain
+    *   value interpolation.
+    */
+  def apply(interpolations: List[Boolean => Res], stringFlagments: List[Either[String, String]]): List[Res] =
+    interleave(interpolations, stringFlagments, List.empty)
+  private[this] def interleave(
+    interpolations: List[Boolean => Res],
+    stringFlagments: List[Either[String, String]],
+    acc: List[Res]
+  ): List[Res] =
+    (stringFlagments, interpolations) match
+      case (Nil, Nil) => acc
+      // this should be always query interpolations because if there is a plain value interpolation, there should be `#` prefix.
```

Comment:
Please move the comment under the `case` block for clarity.

Reply:
Done.